lib/cfa/placer.rb

```diff
@@ -1,6 +1,20 @@
 module CFA
-  # allows to place element at the end of configuration. Default one.
-  class AppendPlacer
+  # Places a new {AugeasElement} into an {AugeasTree}.
+  # Children of this abstract class implement different ways **where**
+  # to place the entry.
+  class Placer
+    # @param  [AugeasTree] tree
+    # @return [AugeasElement,Hash] the new element; it is empty!
+    #   Note that the return value is actually a Hash; {AugeasElement}
+    #   documents its structure.
+    def new_element(tree)
+      abstract_method(tree)
```

Comment:
Yes, that is a weak point. What are the alternatives?

1. This is what we do in lib/installation/proposal_client.rb:
    ```rb
    def new_element(_tree)
      raise NotImplementedError "Calling abstract method 'new_element'"
    end
```

    I wanted to clearly mark the intent, while also not cluttering the documentation with an "unused parameter underscore" like in `_tree`. Actually in the proposal library some docs has disappeared from the rendered YARD because we have documented an argument to the abstract method, named `opts`, but meanwhile renamed it to `_opts` because of RuboCop.

2.  Use https://github.com/openSUSE/abstract_method
    ```rb
require "abstract_method"

abstract_method :new_element
```

    But this loses the param and return documentation, unless we resort to YARD `[@!method](http://www.rubydoc.info/gems/yard/file/docs/Tags.md#method)`

3. Use abstract_method.gem and extend it to allow also an instance variant, `abstract_method(*_args)`